langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/AnthropicChatModelIT.java

```diff
@@ -206,6 +208,42 @@ void should_respect_stop_sequences() {
         assertThat(response.finishReason()).isEqualTo(OTHER);
     }
 
+    @Test
+    void test_all_parameters_with_cache() {
+        AnthropicCacheType cacheType = AnthropicCacheType.builder()
+                .cacheType(AnthropicCacheType.CacheType.EPHEMERAL)
+                .messageTypeApplyCache(AnthropicCacheType.MessageTypeApplyCache.ALL)
+                .build();
+
+        // given
+        ChatLanguageModel model = AnthropicChatModel.builder()
+                .baseUrl("https://api.anthropic.com/v1/")
+                .apiKey(System.getenv("ANTHROPIC_API_KEY"))
+                .version("2023-06-01")
+                .beta("prompt-caching-2024-07-31")
+                .modelName(CLAUDE_3_HAIKU_20240307)
+                .temperature(1.0)
+                .topP(1.0)
+                .topK(1)
+                .maxTokens(100)
+                .stopSequences(asList("hello", "world"))
+                .timeout(Duration.ofSeconds(30))
+                .maxRetries(1)
+                .logRequests(true)
+                .logResponses(true)
+                .cacheType(cacheType)
+                .build();
+
+        SystemMessage systemMessage = SystemMessage.from("You are a professional translator");
+        UserMessage userMessage = new UserMessage(TextContent.from("How say: java is very slow on spanish ?"));
+
+        // when
+        Response<AiMessage> response = model.generate(systemMessage, userMessage);
+
+        // then
+        assertThat(response.content().text()).isNotBlank();
```

Comment:
This test does not really test that caching works.
It would also be great to test tool/user/system messages separately and in combination.

Reply:
This test was really bad, I'll improve it and add tests for error scenarios